ls1intum/artemis-maven-docker
# Enhancement: Install Gradle from Wrapper files

Within these changes, Gradle is installed from the wrapper files in order to cache the Gradle installation in the Docker container to prevent downloading Gradle every time a new build starts.
In addition, Ares has been updated to the newest version including adding the required enforcer rules.

## Steps for Testing
1. Open a Build Plan in Bamboo that builds a Gradle project (optionally with SCA). [This](https://bamboo.ase.in.tum.de/browse/ATESTGRADLETEMPLATE-SOLUTION) is an example you can use.
2. Go to "Actions" (top right) -> "Configure Plan" -> "Default Job" -> "Docker" and set the "Docker image" to "ls1tum/artemis-maven-template:java17-3". Click "Save" and run the build plan
3. Open the logs and verify that you see a log message equal to this:
<img width="1107" alt="Bildschirmfoto 2022-03-31 um 15 31 19" src="https://user-images.githubusercontent.com/73833780/161066599-d2649ee1-c283-4019-81c8-4d6682bf855e.png">

This means that the Gradle installation is not cached and downloaded newly for every build. This PR changes this.
5. Change the "Docker image" to "oleve312/artemis-maven-template:result" and save. This image is the exact same image that would have been generated by the GitHub action for the last commit 32e784d.
6. Rerun the build plan.
7. Verify that the log message from before does not appear anymore. Instead, you should see something like this:
<img width="856" alt="Bildschirmfoto 2022-03-31 um 15 32 01" src="https://user-images.githubusercontent.com/73833780/161066776-bb30d406-ed6d-4596-b788-2fd6011da152.png">
8. Check that a build plan building a Maven project also works as expected with "oleve312/artemis-maven-template:result". [This](https://bamboo.ase.in.tum.de/browse/ATESTNEWDOCKERMAVEN-SOLUTION) is an example for which the settings have already been updated.
